s3datarepository/s3datarepository.go

```diff
@@ -294,9 +307,10 @@ func (s *S3Repository) Provision(ctx context.Context, id string, datasetTags []*
 
 	// create s3 bucket
 	log.Debugf("creating s3 bucket: %s", name)
-	if _, err = s.S3.CreateBucketWithContext(ctx, &s3.CreateBucketInput{
+	_, err = s.S3.CreateBucketWithContext(ctx, &s3.CreateBucketInput{
 		Bucket: aws.String(name),
-	}); err != nil {
+	})
+	if err != nil {
```

Comment:
why did you undo all of these? 😆

Reply:
`err` wasn't getting set properly, breaking the rollback

Reply:
it should, its in scope unless you had a `:=` but I don't see one.

Reply:
i think the new block that i added (for enabling the logging) failed at one point and it didn't run the rollback, and after i changed it to a separate `if` it worked. I then went back and changed all of them assuming they were all broken because the `err` was only scoped to the `if` statement. It's kinda tricky to test that stuff

Reply:
yeah, I resorted to adding errors in the blocks to test the rollback stuff.  I think these should be the condensed blocks though, so we should figure it out.  Otherwise looks good.

Reply:
ok, changed it back and it seems to roll back on error 🤷 maybe i had a `:=` initially